helix-core/src/main/java/org/apache/helix/manager/zk/client/HelixZkClient.java

```diff
@@ -35,8 +36,70 @@
 
   void unsubscribeDataChanges(String path, IZkDataListener listener);
 
+  /* TODO: remove below default implementation when getting rid of I0Itec in the new zk client. */
+  default void subscribeStateChanges(
+      final org.apache.helix.manager.zk.zookeeper.IZkStateListener listener) {
+    subscribeStateChanges(new IZkStateListener() {
```

Comment:
The new anonymous class objects are duplicate code. Can we have the class defined as a private class here so as to avoid duplication?